.github/workflows/ci.yml

```
@@ -268,27 +277,28 @@ jobs:
       - name: Setup Dependencies Windows
         # if a dependency needs to be pinned, see https://github.com/LizardByte/build-deps/pull/186
         if: runner.os == 'Windows'
-        uses: msys2/setup-msys2@v2
+        uses: msys2/setup-msys2@v2.26.0
         with:
-          msystem: ucrt64
-          update: true
+          msystem: ${{ matrix.msystem }}
+          # update: true
           install: >-
             diffutils
             git
             make
             pkg-config
-            mingw-w64-ucrt-x86_64-binutils
-            mingw-w64-ucrt-x86_64-cmake
-            mingw-w64-ucrt-x86_64-gcc
-            mingw-w64-ucrt-x86_64-make
-            mingw-w64-ucrt-x86_64-nasm
-            mingw-w64-ucrt-x86_64-ninja
-            mingw-w64-ucrt-x86_64-onevpl
+            mingw-w64-${{ matrix.toolchain }}-binutils
+            mingw-w64-${{ matrix.toolchain }}-cmake
+            mingw-w64-${{ matrix.toolchain }}-gcc
+            mingw-w64-${{ matrix.toolchain }}-make
+            mingw-w64-${{ matrix.toolchain }}-nasm
+            mingw-w64-${{ matrix.toolchain }}-ninja
+            mingw-w64-${{ matrix.toolchain }}-onevpl
+            mingw-w64-${{ matrix.toolchain }}-perl
 
       - name: Debug msys2.CMD
         if: runner.os == 'Windows'
         run: |
-          cat /d/a/_temp/setup-msys2/msys2.CMD
+          cat "${{ runner.temp }}/setup-msys2/msys2.CMD"
```

```suggestion
          cat "${{ runner.temp }}/setup-msys2/msys2.CMD"
          echo "MSYS2_PATH=${{ runner.temp }}/msys64" >> $GITHUB_ENV
```

This can simplify the detection logic in the CMakeLists.txt. Then just check for this env variable and if it's not set go with the default. It would also self built versions to specify the path without knowing about the Github runner env variables.

---

Done

---

```cmake
set(MSYS2_PATH "C:/msys64")
if(WIN32 AND DEFINED ENV{RUNNER_TEMP})
    set(MSYS2_PATH "$ENV{RUNNER_TEMP}/msys64")
    message(STATUS "Detected CI, get msys2 path from environment: $ENV{RUNNER_TEMP}/msys64")
endif()
```

This will also need to be updated to read from the MSYS2_PATH env variable now.

---

Done :D

---

MSYS2_PATH was taken by MSYS2 shell.
I replace it to MSYS2_ROOT